Sources/UnleashProxyClientSwift/UnleashProxyClientSwift.swift

```diff
@@ -99,14 +94,15 @@ public class UnleashClientBase {
         }
     }
 
-    public func updateContext(context: [String: String]) -> Void {
-        var newContext: [String: String] = [:]
-        newContext["appName"] = self.context["appName"]
-        newContext["environment"] = self.context["environment"]
-
-        context.forEach { (key, value) in
-            newContext[key] = value
-        }
+    public func updateContext(context: [String: String], properties: [String:String]? = nil) -> Void {
+        var newContext = Context(
+            appName: self.context.appName,
+            environment: self.context.environment,
+            userId: context["userId"],
+            sessionId: context["sessionId"],
+            remoteAddress: context["remoteAddress"],
+            properties: properties
+        )
```

Comment:
Because we're still accepting an arbitrary string map, won't this potentially break users' setups? If they were previously calling this function as:
```swift
updateContext([
    "appName": "my-app",
    "custom-property": "what happens to me?"
])
```

Does `custom-property` just get dropped? Previously it would have been added on to the context.

There's also the thing where if you provided `appName` or `environment`, then previously they'd get overwritten. That might be a bug, so I don't mind that not working anymore, but it is something to be aware of.

Reply:
And that's my biggest concern. What should we do in the example you provided?
* it currently works with the Frontend API
* it currently doesn't work with Edge
We can copy those to the custom properties. Which means you don't need extra properties in the overloaded method signature. But if we want to be forgiving then we should accept both context custom properties and explicit properties in the second parameter.

Reply:
> We can copy those to the custom properties. Which means you don't need extra properties in the overloaded method signature. But if we want to be forgiving then we should accept both context custom properties and explicit properties in the second parameter.

I think this is the way to go. Urge people towards the right way to use it, but be forgiving (and in this case backwards compatible) for people using the old way.

Reply:
Added a test describing this behavior